src/vehicle.cpp

```diff
@@ -4255,9 +4257,39 @@ int vehicle::get_z_change() const
     return requested_z_change;
 }
 
-bool vehicle::would_prevent_flyable( const vpart_info &vpinfo ) const
+bool vehicle::would_install_prevent_flyable( const vpart_info &vp, Character &pc ) const
+{
+    if( flyable && !rotors.empty() && !( vp.has_flag( "SIMPLE_PART" ) ||
```

Comment:
You use `vp.has_flag()` here, and `vp.info().has_flag()` below.  It looks like the call to `vp.info()` is unnecessary but you should pick one idion and stick with it, please.

Reply:
It wasn't clear, I changed the `repair` and `remove` function parameters to accept the actual specific `vehicle_part` being repaired/removed rather than the generic `vpart_info` it used to accept before. `install` however still uses `vpart_info` as before, since the part isn't on the vehicle yet.
From what I can tell `vehicle_part.has_flag` doesn't check for JSON flags, it's just a method with the same name. `vpart_info.has_flag` is the method I want to be using in all cases, so in the case of `repair` and `remove` I need to hit `vehicle_part.info().has_flag()` to check the JSON flag.

To make it clearer, I've changed the name of the parameter on the install method so it's obvious.

Reply:
Okay, that works.  Thanks for the explanation.